ui/src/pages/register.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { registerUser } from '../redux/actions';
+
+
+import '../styles/global.css';
+import '../styles/register.css';
+
+const RegisterPage = () => {
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [cpassword, setCPassword] = useState('');
+    const registerError = useSelector((state) => state.errors.registerError);
+    const isLoading = useSelector((state) => state.auth.isLoading);
+    const dispatch = useDispatch();
+
+    const fieldValidation = () => {
+      document.getElementById('error-username').innerText = '';
+      document.getElementById('error-password').innerText = '';
+      document.getElementById('error-compare').innerText = '';
+      let errored = false;
+      if (username.length < 3){
+        document.getElementById('error-username').innerText = 'Username should be at least 3 characters';
+        errored = true;
+      }
+      if (password.length < 3){
+        document.getElementById('error-password').innerText = 'Password should be at least 3 characters';
+        errored = true;
+      }
+      if (password.localeCompare(cpassword) !== 0){
+        document.getElementById('error-compare').innerText = 'Passwords should match';
+        errored = true;
+      }
+      return !errored;
+    };
+
+    return(<div className='main-content'>
+      <form onSubmit={(e) => {
+        e.preventDefault();
+        if(fieldValidation()){
```

Comment:
Small suggestion: Can we change this function name to something that makes it more readable and explains the return value more, such fieldsAreValid. ie
```
if (fieldsAreValid()) {
```